pkg/cli/admin/release/extract_tools.go

```diff
@@ -320,20 +328,29 @@ func (o *ExtractOptions) extractCommand(command string) error {
 			missing.Insert(target.Mapping.Image)
 			continue
 		}
-		klog.V(2).Infof("Will extract %s from %s", target.Mapping.From, spec)
-		ref, err := imagereference.Parse(spec)
+		// Always try the registry/repo passed from the user first.  If image not found,
+		// proceed with the release info from the image spec
+		desired, err := imagereference.Parse(o.From)
 		if err != nil {
 			return err
 		}
-		target.Mapping.Image = spec
-		target.Mapping.ImageRef = imagesource.TypedImageReference{Ref: ref, Type: imagesource.DestinationRegistry}
-		if target.AsArchive {
-			willArchive = true
-			target.Mapping.Name = fmt.Sprintf(target.ArchiveFormat, releaseName)
-			target.Mapping.To = filepath.Join(dir, target.Mapping.Name)
-		} else {
-			target.Mapping.To = filepath.Join(dir, target.Command)
-			target.Mapping.Name = fmt.Sprintf("%s-%s", target.OS, target.Command)
+		digestTag, err := imagereference.GetSpecDigestTag(spec)
+		if err != nil {
+			return err
+		}
+		specFirst := fmt.Sprintf("%s%s", desired.AsRepository(), digestTag)
```

Comment:
@dmage this is where the "@" comes in (look at `oc adm release info anyrelease | grep baremetal-installer`  -or grep any component)

Reply:
i'm updated here to add the "@", removed it from library-go